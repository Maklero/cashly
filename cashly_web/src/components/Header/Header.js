import styled from "styled-components";

const StyledContainer = styled.div`
    width: 100%;
    height: auto;
    padding: 1rem 5rem 1rem 5rem;
    border-bottom: 1px solid ${({ theme }) => theme.colors.grey};
`;

const StyledTextLogo = styled.p`
    font-size: ${({ theme }) => theme.font.sizes.h3};
    font-weight: ${({ theme }) => theme.font.weights.bold};
`;

export default function Header() {
    return (
        <StyledContainer>
            <StyledTextLogo>Cashly</StyledTextLogo>
        </StyledContainer>
    );
}