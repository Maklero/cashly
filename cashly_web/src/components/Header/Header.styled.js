import styled from 'styled-components';

const Header = styled.header`
  height: auto;
  padding: 64px;
  background-color: white;
  border-bottom: 1px solid ${({ theme }) => theme.colors.gray400};
`;

const Wrapper = styled.div`
  display: flex;
  margin: 0 auto;
  max-width: 1024px;
  align-items: center;
  justify-content: space-between;
`;

const Description = styled.p`
  margin-top: 4px;
  color: ${({ theme }) => theme.colors.gray600};
  font-size: ${({ theme }) => theme.fontSizes.h5};
`;

const LeftContentWrapper = styled.div`
  display: flex;
  margin-right: 32px;
  flex-direction: column;
`;

const RightContentWrapper = styled.div`
  display: flex;
  row-gap: 16px;
  margin-left: 8px;
  flex-direction: column;
`;

export {
  Header,
  Wrapper,
  Description,
  LeftContentWrapper,
  RightContentWrapper,
};
