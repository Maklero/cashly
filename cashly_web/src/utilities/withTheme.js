import { ThemeProvider } from "styled-components";
import { theme } from "../styles";

const withTheme = (Component) => {
    return (
        <ThemeProvider theme={theme}>
            <Component />
        </ThemeProvider>
    );
}

export default withTheme;