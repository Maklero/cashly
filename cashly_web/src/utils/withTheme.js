import { ThemeProvider } from 'styled-components'

import { theme } from '@/styles'

export const withTheme = (component) => (
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
)
