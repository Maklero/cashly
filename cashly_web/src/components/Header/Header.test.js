import { render, screen } from "@testing-library/react";
import Header from "./Header";
import withTheme from "../../utilities/withTheme";

test("renders header component", () => {
    const { container } = render(withTheme(Header));

    const headerElm = screen.getByText("Cashly");

    expect(headerElm).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
});