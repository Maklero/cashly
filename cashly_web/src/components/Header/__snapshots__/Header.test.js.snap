// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header component 1`] = `
<div
  class="sc-bdvvtL cccfRp"
>
  <p
    class="sc-gsDKAQ yXxFJ"
  >
    Cashly
  </p>
</div>
`;
